### DISCUSSION TITLE
Implications for online editors?

### DISCUSSION THREAD

**Discussion by JDMCreator** [latex3/tagging-project#745] [Q&A]
I'm running a fairly popular online WYSWYG LaTeX table editor ([website](https://www.latex-tables.com/), [GitHub](https://github.com/JDMCreator/LaTeXTableEditor)). My editor generates code using `tabular`, `longtable`, `tblr` and `longtblr` environments. While I understand the purpose of this project is that tagging would be done automatically, I'm wondering if there would/should be any changes to the code generated by my editor to be compatible with accessible PDFs and `tagpdf`?

**Comment by u-fischer** [comment]
Well there are number of points to consider 

* not every table related package is already fully compatible. tabularray doesn't support tagging at all (issue #177), longtable doesn't work with pdflatex if there are page breaks (issue #22) and so on. So people need to check the current tagging status of external packages. 

* *real content*  (e.g. text) inserted between columns e.g. with `@{text}` is not tagged, so the editor should avoid that.

* Tagged tables should have headers but (with the exception of longtable) there are no commands in tables that declare headers and so the code can not detect which rows or columns to tag with `TH`. Your editor would have to ask the user and add something like `\tagpdfsetup{table/header-rows={1,2},table/header-columns=1}` to setup the first rows and the first column as headers (the columns key will be available in the next release).  The interfaces and key names are not stable yet. At some time we will e.g. extend the optional argument of tabular to allow such settings. See also #1 
There is no support for complex header structures yet, users that want tagged tables should at best stick to headers at the top and left.

* In the next release it will be possible to declare "multirow cells" by putting `\tagpdfsetup{table/multirow={⟨number of rows⟩}` into the top cell. In this case the editor must ensure that the content is in the top cell and that the spanned cells are really empty. Again the interface can still change.
